subprojects/docs/src/docs/userguide/working_with_files.adoc

```diff
@@ -537,23 +478,19 @@ Alternatively, you can remove all doubt by explicitly using link:{javadocPath}/o
 
 Here are some examples of the different types of arguments that the `source` property can take:
 
-=== Example: Specifying a set of files
-
-[source.multi-language-sample,groovy]
-.build.gradle
-----
-include::{samplesPath}/userguide/files/inputFiles/build.gradle[tag=set-input-files]
-----
+.Specifying a set of files
+====
+include::sample[dir="userguide/files/inputFiles/groovy",files="build.gradle[tags=set-input-files]"]
+include::sample[dir="userguide/files/inputFiles/kotlin",files="build.gradle.kts[tags=set-input-files]"]
+====
```

Comment:
:x: the prose above this sample needs to be adjusted

I would change:
```adoc
This means you can, for example, set the property to a `File`, `String`, collection, `FileCollection` or even a closure.
```

to:
```adoc
This means you can, for example, set the property to a `File`, `String`, collection, `FileCollection` or even a closure or `Provider`.
```

Adding `Provider` to the list.